examples/layertree.js

```diff
@@ -100,7 +126,12 @@ app.LayertreeController.prototype.getLayer = function(node) {
  * @export
  */
 app.LayertreeController.prototype.onButtonClick = function(node, layer) {
-  window.alert(node['name'] + ', ' + layer.get('type'));
+  this.http_.get('data/metadata.html').then(angular.bind(this, function(resp) {
+    var html = this.sce_.trustAsHtml(resp.data);
+    this.infoPopup_.setContent(html);
+  }));
```

Comment:
In that specific case it does not make sense to trigger the same request on a each click. You should rely on a the promise returned by `then` instead. Something like this:

```js
app.LayertreeController.prototype.onButtonClick = function(node, layer) {
  if (goog.isNull(this.metadataPromise_)) {
    this.metadataPromise_ = this.http_.get('data/metadata.html').then(function(resp) {
      var html = this.sce_.trustAsHtml(resp.data);
      return html;
    });
  }
  var infoPopup = this.infoPopup_;
  this.metadataPromise_.then(function(html) {
    infoPopup.setTitle(node['name']);
    infoPopup.setContent(html);
    infoPopup.show();
  });
};
```

Does it make sense?